extension/js/common/core/const.ts

```diff
@@ -3,7 +3,7 @@
 'use strict';
 
 export const VERSION = '[BUILD_REPLACEABLE_VERSION]';
-export const FLAVOR: 'consumer' | 'enterprise' = '[BUILD_REPLACEABLE_FLAVOR]' as any;
+export const FLAVOR: 'consumer' | 'enterprise' = '[BUILD_REPLACEABLE_FLAVOR]' as unknown as 'consumer' | 'enterprise';
```

Comment:
I think we don't need `as unknown` here.
`as 'consumer' | 'enterprise'` will be enough

Reply:
It wouldn't work because `'[BUILD_REPLACEABLE_FLAVOR]'` is not of that type. It's a bit of a special case.

Reply:
It works...
Tslint and eslint doesn't return any error and vscode displays fine result.

![image](https://user-images.githubusercontent.com/77344191/178992318-a35e489a-d022-4cf4-8329-a4f09bfb001c.png)

Reply:
This way works because left type is string union type and right type is string.

Reply:
Ah - good. Can be done separately. Thanks for noticing.